### ISSUE TITLE
MontePython can't see .clik files

### ISSUE THREAD

**Issue by DhayaaAnbajagane** [brinckmann/montepython_public#60]
Hi,

I've got MontePython up and running, and have been able to run and analyze JLA data, but have been unable to run MontePython on any of the Planck 2015 Data. I keep getting a likelihood error saying the path to the clik file wasn't found.

This is the command I ran:

python montepython/MontePython.py run -p input/base2015.param -o chains/planck/planck2015

and here is the error that I receive:

-----------------------------------------------------------

Traceback (most recent call last):
  File "montepython/MontePython.py", line 40, in <module>
    sys.exit(run())
  File "/home/dhayaa/Programs/Codes/montepython_public-master/montepython/run.py", line 31, in run
    custom_command)
  File "/home/dhayaa/Programs/Codes/montepython_public-master/montepython/run.py", line 190, in safe_initialisation
    cosmo, data, command_line, success = initialise(custom_command)
  File "/home/dhayaa/Programs/Codes/montepython_public-master/montepython/initialise.py", line 66, in initialise
    data = Data(command_line, path)
  File "/home/dhayaa/Programs/Codes/montepython_public-master/montepython/data.py", line 352, in __init__
    self.initialise_likelihoods(self.experiments)
  File "/home/dhayaa/Programs/Codes/montepython_public-master/montepython/data.py", line 481, in initialise_likelihoods
    elem, elem, folder, elem)
  File "<string>", line 1, in <module>
  File "/home/dhayaa/Programs/Codes/montepython_public-master/montepython/likelihood_class.py", line 897, in __init__
    " Note that the default path to search for it is"
montepython.io_mp.LikelihoodError:

Likelihood Error:
 /|\   The path to the .clik file for the likelihood Planck_highl was not found
/_o_\  where indicated:
       /home/dhayaa/Programs/Codes/plc-2.1_py3/../plc_2.0/hi_l/plik/plik_dx11dr2
       _HM_v18_TT.clik
       Note that the default path to search for it is one directory above the
       path['clik'] field. You can change this behaviour in all the
       Planck_something.data, to reflect your local configuration, or
       alternatively, move your .clik files to this place.

-----------------------------------------------------------

I've checked all the ways I know how, and the files are indeed at the path indicated in the error message. I have additionally tried this with the plc-2.0 files, thinking maybe the updated plc-2.1 version was causing some issues, but running all this with plc-2.0 didn't solve the issue either.

Could I get some assistance with understanding what is going wrong here? I'd very much appreciate it. Thanks!

**Comment by lukashergt** [comment]
Hi Dhayaa,

what does your `default.conf` file look like? And can you show the path to your clik folder?

I had the problem a while back that there was a very subtle spelling difference which is easily overlooked in a long path string: `plc_2.0` and `plc-2.0`

Is that maybe the problem? Especially since you have some mixed `plc-2.1` and `plc_2.0` in there...

Best,
Lukas

**Comment by DhayaaAnbajagane** [comment]
Hey Lukas,

Thanks for the response! I'm pasting the requested information below.

1) The three uncommented lines in my default.conf file:

---------

root = '/home/dhayaa/Programs/Codes/'

path['cosmo']	= root + '/class_public-2.7.2'
path['clik']    = root + '/plc-2.1_py3'

-------

2) This is the path to my clik folder: /home/dhayaa/Programs/Codes/plc-2.1_py3

3) This is the path to the clik file that MontePython can't see: /home/dhayaa/Programs/Codes/plc_2.0/hi_l/plik/plik_dx11dr2_HM_v18_TT.clik

I've spent a while trying different styles of writing the path to see if something on my end was treating the path differently, but so far that hasn't revealed anything. Also, I'm not sure if this makes a big difference for this problem but---I'm running MontePython on a computing cluster and not from my personal laptop.

Thanks for your time,
Dhayaa

**Comment by lukashergt** [comment]
Hi Dhayaa,

the problem doesn't lie with path specification in the `default.conf` file, but with the different directory names `plc-2.1_py3` (your directory containing the Planck clik files) and `plc_2.0` (the name that MontePython searches for). From the error message in your first post you can see that `path['clik']` sends the code first to the directory `plc-2.1_py3/`. From there it goes one level down `../` and then tries to re-enter the directory but using the wrong (outdated) name `plc_2.0`:
<pre>
 /home/dhayaa/Programs/Codes/<b>plc-2.1_py3</b>/../<b>plc_2.0</b>/hi_l/plik/plik_dx11dr2
</pre>

Check the Planck likelihood `.data` files in e.g. `montepython/likelihoods/Planck_highl/Planck_highl.data`. There it probably reads 
<pre>
Planck_highl.path_clik = os.path.join(data.path['clik'],'../<b>plc_2.0</b>/hi_l/plik/plik_dx11dr2_HM_v18_TT.clik')
</pre>
but the `plc_2.0` would need to be updated to `plc-2.1_py3` in your case. And accordingly for the `lowl` and `lensing` likelihood files.

You mentioned in your first post that you tried this with the previous `plc-2.0` version. I suspect the reason this went wrong is because of the dash/underscore naming difference I mentioned in my previous post.

Best,
Lukas

**Comment by DhayaaAnbajagane** [comment]
Hey again Lukas,

Just to make sure I understand: does the planck data have to be placed in the same folder as the planck likelihood code then?

I have one folder named plc-2.1_py3 that contains all the clik libraries/code for computing likelihoods, and a folder named plc_2.0 that has the data for the 6 base datasets. Both these folders are contained within the folder `/Codes`. Nevertheless, I have tried what you suggested in your message:

1. First added hi_l file to the plc-2.1_py3 folder so MontePython could find it there. 
2. Changed Planck_highl.data file so path was what you suggested.

However, the previous error still persisted, but with the new path 

`Likelihood Error:
 /|\   The path to the .clik file for the likelihood Planck_highl was not found
/_o_\  where indicated:
       /home/dhayaa/Programs/Codes/plc-2.1_py3/../plc-2.1_py3/hi_l/plik/plik_dx1
       1dr2_HM_v18_TT.clik
       Note that the default path to search for it is one directory above the
       path['clik'] field. You can change this behaviour in all the
       Planck_something.data, to reflect your local configuration, or
       alternatively, move your .clik files to this place.
`

Thank you for your time,
Dhayaa

**Comment by lukashergt** [comment]
You should not move anything around. MontePython's Planck likelihood files should be in folders such as `montepython/likelihoods/Planck_highl/`. They have to be exactly there. The click folders can be anywhere as long as `default.conf` points correctly to them.

I'm afraid I'm unfamiliar with the `plc-2.1_py3` version, I only know `plc_2.0`. You are saying `plc-2.1_py3` contains clik libraries. Can you verify that `/home/dhayaa/Programs/Codes/plc-2.1_py3/hi_l/plik/plik_dx1 1dr2_HM_v18_TT.clik` exists?

Else, have you tried installing the `plc_2.0` version following the instructions in the README? Especially the last bit which seems to go wrong in your case:

> The four likelihoods defined in |MP| for Planck are Planck_highl, Planck_lowl, Planck_lensing, lowlike (the polarization data from WMAP). In each of the respective data files for these likelihood, please make sure that the line, for instance,
> 
> `Planck_highl.path_clik = data.path['clik']+'../something.clik'`
> 
> points to the correct clik file. Do not forget to source your Planck likelihood every time you want to use it:
> 
> `$ source Your/Plc/bin/clik_profile.sh`
> 
> You can put this line in your .bashrc file, and you should put it in your scripts for cluster computing.

**Comment by DhayaaAnbajagane** [comment]
Hi Lukas,

Apologies, but I'm afraid I'm a bit confused again. The 2015 Planck data which I downloaded from [here](http://pla.esac.esa.int/pla/#cosmology), has a **plc-2.0** file (which contains the MakeFile, the src folder, setupy.py, and others), and then a **plc_2.0** file (which contains the ".clik" files for high_l, lensing data etc.). The **plc-2.1_py3** file I mention is just an update to the plc-2.0 file, and not the plc_2.0 file. Sorry for not being explicit about that before.

I can `cd` to the directory `/home/dhayaa/Programs/Codes/plc-2.1_py3/hi_l/plik/plik_dx1 1dr2_HM_v18_TT.clik` and view it. Is this enough to verify it's existence, or should I do something more robust?

Also, I haven't moved the __init__.py, and the .data files from any of the likelihoods folders (such as `likelihoods/Planck_highl/`). Finally, when I `cd` to the .clik folder and use `$PWD` I get `/home/dhayaa/Programs/Codes/plc_2.1_py3/hi_l/plik/plik_dx11dr2_HM_v18_TT.clik`, which makes me think the full path I have is indeed the correct one.

I have already installed plc-2.0 twice now to no avail so I'm not sure that would solve my issue.

**Comment by DhayaaAnbajagane** [comment]
**_Quick update:_**

I put the data files from JLA in the same directory as the .clik file causing issues (`/home/dhayaa/Programs/Codes/plc-2.1_py3/hi_l/plik/`) and then changed the `JLA.data` file in the `likelihoods/JLA` folder to point it to the new path. When I run the analysis everything works perfectly. This leads me to believe this isn't a path issue.

**Comment by lukashergt** [comment]
You made sure in the `plc-2.0` version to be consistent in the use of dash and underscore between the `default.conf` file and the `.data` files?

If you now can `cd` into the path from the error message, and if you made sure to source the `clik_profile.sh` in your `.bashrc` or your cluster script, then I'm afraid I'm at a loss as well.

Maybe @brinckmann  can be of more help.

**Comment by DhayaaAnbajagane** [comment]
Hi Lukas,

After reinstalling plc-2.0 a bunch of times, MontePython finally works with the Planck data! I'm honestly not sure what broke or what fixed it. I changed absolutely nothing during the many installations I tried, but it seems to have worked now, so fingers crossed that it stays working. Thanks for persisting with this issue and sharing your time and effort. I appreciate it! :)

Thanks,
Dhayaa
